src/catalog/src/memory/objects.rs

```diff
@@ -91,6 +91,7 @@ pub struct Schema {
     pub oid: u32,
     pub items: BTreeMap<String, GlobalId>,
     pub functions: BTreeMap<String, GlobalId>,
+    pub types: BTreeMap<String, GlobalId>,
```

Comment:
Not for this PR, but should we be doing something similar to all non-relation items? Like secrets and connections?

Reply:
See the new big ol' comment on `CatalogItemType::conflicts_with_type`. The tl;dr is that only functions and types are special, and everything else should get intermixed in the main "items" namespace.